store/hbase/txn.go

```diff
@@ -0,0 +1,235 @@
+		}
+
+		return step, nil
+	}
+
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	intVal, err := strconv.ParseInt(string(val), 10, 0)
+	if err != nil {
+		return intVal, errors.Trace(err)
+	}
+
+	intVal += step
+	err = txn.UnionStore.Set(k, []byte(strconv.FormatInt(intVal, 10)))
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	return intVal, nil
+}
+
+func (txn *hbaseTxn) String() string {
+	return fmt.Sprintf("%d", txn.tID)
+}
+
+func (txn *hbaseTxn) Get(k kv.Key) ([]byte, error) {
+	log.Debugf("get key:%q, txn:%d", k, txn.tID)
+	k = kv.EncodeKey(k)
+	val, err := txn.UnionStore.Get(k)
+	if kv.IsErrNotFound(err) || len(val) == 0 {
+		return nil, kv.ErrNotExist
+	}
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return val, nil
+}
+
+// GetInt64 get int64 which created by Inc method.
+func (txn *hbaseTxn) GetInt64(k kv.Key) (int64, error) {
+	k = kv.EncodeKey(k)
+	val, err := txn.UnionStore.Get(k)
+	if kv.IsErrNotFound(err) {
+		return 0, nil
+	}
+
+	if err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	intVal, err := strconv.ParseInt(string(val), 10, 0)
+	return intVal, errors.Trace(err)
+}
+
+func (txn *hbaseTxn) Set(k kv.Key, data []byte) error {
+	if len(data) == 0 {
+		// Incase someone use it in the wrong way, we can figure it out immediately
+		debug.PrintStack()
+		return ErrCannotSetNilValue
+	}
+
+	log.Debugf("set key:%q, txn:%d", k, txn.tID)
+	k = kv.EncodeKey(k)
+	return txn.UnionStore.Set(k, data)
+}
+
+func (txn *hbaseTxn) Seek(k kv.Key) (kv.Iterator, error) {
+	log.Debugf("seek %q txn:%d", k, txn.tID)
+	k = kv.EncodeKey(k)
+	iter, err := txn.UnionStore.Seek(k, txn)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return iter, nil
+}
+
+func (txn *hbaseTxn) Delete(k kv.Key) error {
+	log.Debugf("delete %q txn:%d", k, txn.tID)
+	k = kv.EncodeKey(k)
+	err := txn.UnionStore.Delete(k)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (txn *hbaseTxn) each(f func(iterator.Iterator) error) error {
+	iter := txn.UnionStore.Dirty.NewIterator(nil)
+	defer iter.Release()
+	for iter.Next() {
+		if err := f(iter); err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+func (txn *hbaseTxn) doCommit() error {
+	err := txn.each(func(iter iterator.Iterator) error {
+		var row, val []byte
+		row = make([]byte, len(iter.Key()))
+		if len(iter.Value()) == 0 { // Deleted marker
+			copy(row, iter.Key())
+			d := hbase.NewDelete(row)
+			d.AddStringColumn(ColFamily, Qualifier)
+			txn.Txn.Delete(txn.storeName, d)
+		} else {
+			val = make([]byte, len(iter.Value()))
+			copy(row, iter.Key())
+			copy(val, iter.Value())
+			p := hbase.NewPut(row)
+			p.AddValue([]byte(ColFamily), []byte(Qualifier), val)
+			txn.Txn.Put(txn.storeName, p)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = txn.Txn.Commit()
+	if err != nil {
+		log.Error(err)
+		return errors.Trace(err)
+	}
+
+	txn.version = kv.NewVersion(txn.Txn.GetCommitTS())
```

Comment:
the kv has a interface naming GetCommittedVersion, so I though we should use CommittedTS.